packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_NullspaceFactory_def.hpp

```diff
@@ -146,8 +190,47 @@ namespace MueLu {
               nsValues[j*numPDEs + i] = 1.0;
             }
           }
+          if ( (int) nullspaceDim > numPDEs ) {
+            /* xy rotation */
+            ArrayRCP<Scalar> nsValues = nullspace->getDataNonConst(numPDEs);
+            int numBlocks = nsValues.size() / numPDEs;
+            TEUCHOS_TEST_FOR_EXCEPTION(numBlocks != (int) xvals.size(), Exceptions::RuntimeError, "MueLu::NullspaceFactory::Build(): number of coordinates does not match  ndofs/numPDEs.");
+            for (int j=0; j< numBlocks; ++j) {
+              nsValues[j*numPDEs + 0] = -(yvals[j]-cy);
+              nsValues[j*numPDEs + 1] =  (xvals[j]-cx);
+            }
+          }
+          if ( (int) nullspaceDim == numPDEs+3 ) {
+            /* xz rotation */
+            ArrayRCP<Scalar> nsValues = nullspace->getDataNonConst(numPDEs+1);
+            int numBlocks = nsValues.size() / numPDEs;
+            for (int j=0; j< numBlocks; ++j) {
+              nsValues[j*numPDEs + 1] = -(zvals[j]-cz);
+              nsValues[j*numPDEs + 2] =  (yvals[j]-cy);
+            }
+            /* yz rotation */
+            nsValues = nullspace->getDataNonConst(numPDEs+2);
+            numBlocks = nsValues.size() / numPDEs;
+            for (int j=0; j< numBlocks; ++j) {
+              nsValues[j*numPDEs + 0] = -(zvals[j]-cz);
+              nsValues[j*numPDEs + 2] =  (xvals[j]-cx);
+            }
+          }
+           /*
+          // Scale columns to match what Galeri does. Not sure how important this is  as QR factorization
+          // when creating ptent should take care of scaling issues ... but leaving it just in case.
+           if ( (int) nullspaceDim > numPDEs ) {
+            Teuchos::Array<typename Teuchos::ScalarTraits<Scalar>::magnitudeType> norms2(nullspaceDim);
+            nullspace->norm2(norms2);
+            Teuchos::Array<Scalar> norms2scalar(nullspaceDim);
+            for (int i = 0; i < nullspaceDim; i++)
+              norms2scalar[i] = norms2[0] / norms2[i];
+            nullspace->scale(norms2scalar);
+          }
+          */
```

Comment:
```suggestion
```

Reply:
This redundancy was already in the code ... and I just continued it. It basically corresponds to whether or not we have a blockmap or a map. I'll look into whether this can be done in a single function.